source/slang/slang-ir-link.cpp

```diff
@@ -40,9 +40,6 @@ struct IRSpecEnv
 
 struct IRSharedSpecContext
 {
-    // The code-generation target in use
-    CodeGenTarget target;
-
     // The API-level target request
     TargetRequest* targetReq = nullptr;
```

Comment:
Unrelated to this change: are we at a point where we could eliminate the dependency of IR linking on the compilation target? If all target specialization now uses `__target_switch`, we should be able to finally get rid of that source of complexity in linking.

Reply:
this is probalby worth a separate clean up pass, I believe there are still some `__target_intrinsic` modifiers lingering around, particularly for types.